--- ppp-plugin-ifcfg-password-0.1/ifcfg-password.c.orig	2003-01-01 00:00:00.000000000 +0000
+++ ppp-plugin-ifcfg-password-0.1/ifcfg-password.c	2024-01-01 00:00:00.000000000 +0000
@@ -13,9 +13,18 @@

 #include <errno.h>
 #include <unistd.h>
+#include <stdio.h>
+#include <limits.h>
 #include <syslog.h>
 #include <string.h>
+
 #include "pppd.h"
+#include "options.h"
+#include "upap.h"
+
+#if !defined(VERSION) && defined(PPPD_VERSION)
+#define VERSION PPPD_VERSION
+#endif

 char pppd_version[] = VERSION;

@@ -128,6 +137,6 @@

 void plugin_init(void)
 {
-	add_options(options);
+	ppp_add_options(options);
 	pap_passwd_hook = read_password;
 }
